library/ssl_cli.c

```diff
@@ -4361,7 +4361,14 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
            ret = mbedtls_ssl_parse_delayed_certificate_verify( ssl, authmode,
                                    ssl->session_negotiate->peer_cert, rs_ctx );
            if( ret != 0 )
+           {
                break;
+           }
+           if( ret != 0 )
+           {
+               break;
```

Comment:
you can return an FI error in the second if condition.  And it is better to add a comment to explain the double check

Reply:
It wouldn't do much change, since someone would be skipping into an error anyway, but sure, I can add that. And a comment, of course.